crates/bevy_render/src/render_asset.rs

```diff
@@ -235,21 +241,30 @@ pub(crate) fn extract_render_asset<A: RenderAsset>(
         |world, mut cached_state: Mut<CachedExtractRenderAssetSystemState<A>>| {
             let (mut events, mut assets) = cached_state.state.get_mut(world);
 
-            let mut changed_assets = <HashSet<_>>::default();
+            let mut needs_extracting = <HashSet<_>>::default();
             let mut removed = <HashSet<_>>::default();
+            let mut modified = <HashSet<_>>::default();
 
             for event in events.read() {
                 #[expect(
                     clippy::match_same_arms,
                     reason = "LoadedWithDependencies is marked as a TODO, so it's likely this will no longer lint soon."
                 )]
                 match event {
-                    AssetEvent::Added { id } | AssetEvent::Modified { id } => {
-                        changed_assets.insert(*id);
+                    AssetEvent::Added { id } => {
+                        needs_extracting.insert(*id);
+                    }
+                    AssetEvent::Modified { id } => {
+                        needs_extracting.insert(*id);
+                        modified.insert(*id);
```

Comment:
could we not push modified meshes to `needs_extracting` and chain it with `modified` when its used?

Reply:
In this code block, just do `modified.insert(*id)`, then in the loop over `needs_extracting` later do `needs_extracting.chain(modified)`

Reply:
Thinking about this more, it might not be possible because of the drain part?

Reply:
Oh, yeah won't be possible.